webhook-integrations/raygun/notes.md

```diff
@@ -0,0 +1,158 @@
+official Raygun docmentation
+[here][https://raygun.com/docs/integrations/webhooks).*
+
+In the next section we'll be analyzing certain things that one may
+get from a payload.
+
+![](screenshots/5-webhook-display.png)
+
+### Structure of the payload
+
+`event`: Currently known events are `error_notifications` and `error_activity`. 
+The first one came from the application and the second one comes from Raygun
+panel (manually and intentionaly triggered by the user).  
+`eventType`: Type of the event. This varies depending on the `event` that you
+receive. For `error_notifications`, it can be `NewErrorOccurred` until
+`HourlyFollowUp`. For `error_activity`, it can be `StatusChanged`,
+`AssignedToUser`, and `CommentAdded`.  
+`error`: Contains a dictionary detailing the error details. **This name is
+misleading**. It can also contain a description for a Raygun activity such as
+error assignment messages or error status that has changed.
+
+`url`: An URL of the error that redirects to Raygun  
+`message`: Message of the error in notifications/Message of the activity.  
+`status`: The status change for the error. Only for activity.  
+`user`: The user who did the status change. Only for activity.  
+
+`firstOccurredOn`: *self-explanatory*. Only for notifications.  
+`lastOccurredOn`: *self-explanatory*. Only for notifications.  
+`activityDate`: The time this activity occurred. Only for activity.
+`usersAffected`: Number of users affected by this exact error in integer. Only
+for notifications.  
+`totalOccurrences`: Number of times this exact error has ever happened in
+integer. Only for notifications.  
+`instance`: Contains the instance of the error. Only for notification.
+
+`tags`: Error tags.  
+`affectedUser`: The identity of the user affected by the error  
+`Identifier`: Unique identifier of the user
+`IsAnonymous`: true or false  
+`UUID`: Unique identifier of the user. From observation, seems like this value
+is the same with `Identifier`.  
+
+`customData`: Custom data supplied by the error. If there is none, it it
+`null`.
+
+`application`: Contains the data of the application
+
+`name`: Name of the application in Raygun  
+`url`: URL to the application in Raygun
+
+## Possible messages
+
+Based on Slack, the user can expect these types of messages.
+
+Test message:
+
+![](screenshots/1.1-test-message.png)
+
+Exception error messages:
+
+![](screenshots/2.0-exception.png)
+
+Error assigned to user:
+
+![](screenshots/2.1-assigned.png)
+
+Error resolved:
+
+![](screenshots/2.2-resolved.png)
+
+Error reactivated:
+
+![](screenshots/2.3-reactivated.png)
+
+Error ignored:
+
+![](screenshots/2.4-ignore.png)
+
+Repeated error messages:
+
+![](screenshots/3.0-repeated-error.png)
+
+Repeated error resolved:
+
+![](screenshots/3.1-resolved.png)
+
+Repeated error reactivated:
+
+![](screenshots/3.2-reactivated.png)
+
+Repeated error assigned:
+
+![](screenshots/3.3-assigned.png)
+
+Repeated error ignored:
+
+![](screenshots/3.4-ignored.png)
+
```

Comment:
I only have one minor question. The possible messages you list here are a lot more than the number of payloads you submitted as part of your other PR. Do the payloads in that PR cover the possible messages listed here? :)

Reply:
Yes. Take a look at this example:

```{
  "event":"error_activity",
  "eventType":"StatusChanged",
  "error": {
    "url":"https://app.raygun.com/error-url",
    "description":"Emma marked error Script Error as Ignored",
    "activityDate":"1970-01-28T01:49:36Z",
    "message": "Script Error",
    "status":"Ignored",
    "user":"Emma Cat"
  },
  "application": {
    "name":"Best App",
    "url":"http://app.raygun.io/application-url"
  }
}
```

Notice the key `status` here can be changed into `ignored`, `resolved`, etc. So you really only need one fixture to test it.

Reply:
Ah okay makes sense! Thanks, @Privisus! :)